packages/react/src/CloudinaryImage.tsx

```diff
@@ -0,0 +1,95 @@
+import { parseRotate } from './transformationParsers/parseRotate';
+import { parseGravity } from './transformationParsers/parseGravity';
+import { parseAspectRatio } from './transformationParsers/parseAspectRatio';
+import { parseIgnoreAspectRatio } from './transformationParsers/parseIgnoreAspectRatio';
+import { parseZoom } from './transformationParsers/parseZoom';
+import { parseRoundCorners } from './transformationParsers/parseRoundCorners';
+import { parseOpacity } from './transformationParsers/parseOpacity';
+import { Quality } from './transformationTypes/quality';
+import { ImageFormat } from './transformationTypes/format';
+import { RemoveBackground } from './transformationTypes/removeBackground';
+import { Effect } from './transformationTypes/effect';
+import { Background } from './transformationTypes/background';
+import { Rotate } from './transformationTypes/rotate';
+import { RoundCorners } from './transformationTypes/roundCorners';
+import { Opacity } from './transformationTypes/opacity';
+import { parseRemoveBackground } from './transformationParsers/parseRemoveBackground';
+import { Resize } from './transformationTypes/resize';
+
+type ImageTransformationProps = {
+  quality?: Quality;
+  format?: ImageFormat;
+  removeBackground?: RemoveBackground;
+  effects?: Effect[];
+  background?: Background;
+  rotate?: Rotate;
+  roundCorners?: RoundCorners;
+  opacity?: Opacity;
+  resize?: Resize;
+};
+
+type ImageV3Props = {
+  cldImg?: never
+  src: string;
+  alt: string;
+  children?: never;
+} & ImageTransformationProps;
+
+interface ImageV2Props {
+  src?: never;
+  cldImg: UrlGenCloudinaryImage;
+  children?: never;
+}
+
+export type CloudinaryImageProps = ImageV3Props | ImageV2Props;
+
+export const CloudinaryImage = forwardRef<HTMLImageElement, CloudinaryImageProps>(
+  (props, ref) => {
+    if (props.cldImg) {
+      const { cldImg, ...rest } = props;
+      return <img src={cldImg.toURL()} {...rest} ref={ref} />;
+    }
+    const transformationMap = {
+      format: parseFormat,
+      quality: parseQuality,
+      background: parseBackground,
+      removeBackground: parseRemoveBackground,
+      effects: parseEffects(parseRemoveBackground),
+      resize: createParseResize({
+        parseHeight,
+        parseAspectRatio,
+        parseGravity,
+        parseWidth,
+        parseBackground,
+        parseIgnoreAspectRatio,
+        parseZoom
+      }),
+      rotate: parseRotate,
+      roundCorners: parseRoundCorners,
+      opacity: parseOpacity
+    } satisfies TransformationMap<ImageTransformationProps>;
+    const { src, alt, children, cldImg, ...rest } = props;
+    const { baseCloudUrl, assetPath } = parseCloudinaryUrlToParts(src);
+    const transformationString = parsePropsToTransformationString({
+      transformationProps: rest,
+      transformationMap
+    });
+
+    return <img src={`${baseCloudUrl}/${transformationString}/${assetPath}`} alt={alt} ref={ref} />;
```

Comment:
I'm pretty sure we should forward any other props here and allow users pass whatever they want to avoid issue raised that we don't anticipate yet.